test/sanity_check.sh

```diff
@@ -5,23 +5,42 @@
 set -e
 
 whitely echo Ping each host from the other
+
 for host in $HOSTS; do
     for other in $HOSTS; do
-        [ $host = $other ] || run_on $host $PING $other
+        if [ "$host" != "$other" ]; then
+            echo $(run_on $host $PING $other) &
+            pids="$pids $!"
+        fi
     done
 done
+for pid in $pids; do wait $pid; done
+unset pids
```

Comment:
I think I would put this (or `pids=""`) at the beginning of the loop rather than the end.